### PR TITLE
AO3-5942 Allow video embeds from bilibili.com

### PR DESCRIPTION
**Pull request by Rei-Ruimin**
# Pull Request Checklist

* [X] Have you read ["How to write the perfect pull request"](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [X] Have you read the [contributing guidelines](https://github.com/otwcode/otwarchive/blob/master/CONTRIBUTING.md)?
* [X] Have you added [tests for any changed functionality](https://github.com/otwcode/otwarchive/wiki/Automated-Testing)?
* [X] Have you added the [Jira](https://otwarchive.atlassian.net) issue number
  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
* [X] Do you have fewer than 5 pull requests already open? If not, please wait
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-5942

## Purpose
To allow video embeds from https www.bilibili.com
I added bilibili to the WHITELIST_REGEXES of class EmbedSanitizer and added bilibili to the HtmlCleaner test

## Testing Instructions

How can the Archive's QA team verify that this is working as you intended?
By embedding video from bilibili.com with<iframe> tag in html format. 
For example: 
<iframe src="//player.bilibili.com/player.html?aid=209713442&amp;bvid=BV1Bh411x7PH&amp;cid=457490304&amp;page=1" frameborder="no" allowfullscreen="true">

## Credit

Rimi (Rei in Jira) She/her

### CODE REVIEW

**Comment by sarken** [comment]
Hi, Rimi!

Thank you for the pull request -- someone will be along to review it soon.

In the meantime, I've assigned the Jira issue to you and set its status to In Review, so no one will mistakenly create a duplicate pull request. I've also given your account permission to comment on, assign, and transition issues in the future. 

However, it makes things a bit easier for us on the organizational side if the Full Name on your Jira account either closely matches the name you'd like us to credit in the release notes or includes it in parentheses, e.g. "Nickname (PREFERRED NAME).") Is the Full Name something you could change? If not, no worries, we'll find a way to work with it as is.

Thanks again for contributing! If you have any questions, you can contact us at otw-coders@transformativeworks.org.

File: lib/otw_sanitize/embed_sanitizer.rb [hunk 1]
```diff
@@ -18,7 +18,8 @@ class EmbedSanitizer
       viddersnet:       %r{^vidders\.net/},
       viddertube:       %r{^viddertube\.com/},
       vimeo:            %r{^(player\.)?vimeo\.com/},
-      youtube:          %r{^youtube(-nocookie)?\.com/}
+      youtube:          %r{^youtube(-nocookie)?\.com/},
+      bilibili:         %r{^(player\.)?bilibili\.com/}
```
Comment (redsummernight) [inline lib/otw_sanitize/embed_sanitizer.rb:22]: Can you keep the list alphabetized?

**Comment by redsummernight** [comment]
Since the site supports HTTPS, can you add it to

https://github.com/otwcode/otwarchive/blob/cbb45213e7ab83fcfd84a584e79dd7abcf7df720/lib/otw_sanitize/embed_sanitizer.rb#L29-L32

and update the corresponding test at https://github.com/otwcode/otwarchive/blob/cbb45213e7ab83fcfd84a584e79dd7abcf7df720/spec/lib/html_cleaner_spec.rb#L22-L31?

**Comment by Rei-Ruimin** [comment]
Sure! I just added the changes

File: lib/otw_sanitize/embed_sanitizer.rb [hunk 2]
```diff
@@ -28,7 +29,7 @@ class EmbedSanitizer
 
     SUPPORTS_HTTPS = %i[
       ao3 archiveorg eighttracks podfic
-      soundcloud spotify viddersnet viddertube vimeo youtube
+      soundcloud spotify viddersnet viddertube vimeo youtube bilibili
```
Comment (redsummernight) [inline lib/otw_sanitize/embed_sanitizer.rb:32]: Just one more comment, can you keep this list sorted as well?

Comment (Rei-Ruimin) [reply lib/otw_sanitize/embed_sanitizer.rb]: done :D

**Review by redsummernight — APPROVED** [review]
Looks good, thank you!

**Pull request merged**